vast_csi/plugins/base.py

```diff
@@ -352,14 +362,23 @@ def _get_publish_context_for_non_attach_required(
         return resp.publish_context
 
 
-    def _store_meta_file(self, meta_file, volume_id, is_ephemeral, vms_session):
+    def _store_meta_file(self, meta_file, volume_id, is_ephemeral, vms_session, is_host_encryption=False):
         """
         Stores metadata about a volume in a file, including information about
-        whether the volume is ephemeral and, if so, serialized session data.
+        whether the volume is ephemeral, host_encryption and, if so, serialized session data.
         """
-        payload = dict(volume_id=volume_id, is_ephemeral=is_ephemeral)
+        payload = {
+        "volume_id": volume_id,
+        "is_ephemeral": is_ephemeral,
+        "is_host_encryption": is_host_encryption,
+        }
+
         if is_ephemeral:
             payload["vms_session"] = vms_session.serialize(salt=volume_id)
+
+        if is_host_encryption:
```

Comment:
if ephemeral or is_host_entryption:  ....

Reply:
I didn't get why "vms_session" should be serialized in is_host_encryption is True?

It is limitation of ephemeral volumes that secrets are not passed via context.  So we need to serialize session for deletion of such volumes.

Reply:
Ah OK, thanks for catching! Will fix.